File: 0509-fibonacci-number/0509-fibonacci-number.py
class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        if(n<=1) :
            return n

        first = 0
        second = 1
        current = 0

        for i in range(2,n+1) :
            current = first + second 
            first = second 
            second = current

        return current
        

File: 0509-fibonacci-number/0509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        if (n <= 1) {
            return n;
        }
        
        int first = 0;
        int second = 1;
        int current = 0;
        
        for (int i = 2; i <= n; i++) {
            current = second + first;
            first = second;
            second = current;
        }
        
        return current;
    }
}